nerfstudio/model_components/renderers.py

```diff
@@ -101,19 +101,15 @@ def combine_rgb(
         else:
             comp_rgb = torch.sum(weights * rgb, dim=-2)
             accumulated_weight = torch.sum(weights, dim=-2)
-        if BACKGROUND_COLOR_OVERRIDE is not None:
```

Comment:
Thanks @Mxbonn!

Can you double-check this logic? I think the override should be applied even when the background color is set to `random` or `last_sample`, which doesn't seem like it would happen if you remove this first `if` statement.

Adding the `else` at the end makes sense to me though.

Reply:
Hi @brentyi, you're right. It should indeed still be applied in case of an override.
However, the current way to do it seems a bit awkward to me since get_background_color serves both to override the background color and to convert it to RGB. But you cannot just simply pass background color `random` or `last_sample` because this will fail the assertion in case there is no override. So you need to do the check for override before calling the method. But if you're doing the override check anyway, then you should immediately apply it instead of calling the get_background_color.
Anyways I made a change to the code which should be correct now.